### PR TITLE
Multiple chart fixes

### PR DESCRIPTION
**Pull request by Pierre-Gilles**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- [x] If your changes affects code, did your write the tests?
- [x] Are tests passing? (`npm test` on both front/server)
- [x] Is the linter passing? (`npm run eslint` on both front/server)
- [x] Did you run prettier? (`npm run prettier` on both front/server)
- [x] If you are adding a new features/services, did you run integration comparator? (`npm run compare-translations` on front)
- [x] Did you test this pull request in real life? With real devices? If this development is a big feature or a new service, we recommend that you provide a Docker image to [the community](https://community.gladysassistant.com/) for testing before merging.
- [x] If your changes modify the API (REST or Node.js), did you modify the API documentation? (Documentation is based on comments in code)
- [x] If you are adding a new features/services which needs explanation, did you modify the user documentation? See [the GitHub repo](https://github.com/GladysAssistant/v4-website) and the [website](https://gladysassistant.com).
- [x] Did you add fake requests data for the demo mode (`front/src/config/demo.js`) so that the demo website is working without a backend? (if needed) See [https://demo.gladysassistant.com](https://demo.gladysassistant.com).

NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open.

### Description of change

Screenshot 2021-11-11 at 13 29 13

Screenshot 2021-11-11 at 14 16 59

### CODE REVIEW

File: front/src/components/boxs/chart/Chart.jsx [hunk 1]
```diff
@@ -155,7 +161,15 @@ class Chartbox extends Component {
...
-      if (data.length > 0) {
+      // Before now, there was a "unit" attribute in this box instead of "units",
+      // so we need to support "unit" as some users may already have the box with that param
+      const unit = this.props.box.units ? this.props.box.units[0] : this.props.box.unit;
+      // We check if all deviceFeatures selected are in the same unit
+      const allUnitsAreSame = this.props.box.units ? allEqual(this.props.box.units) : false;
+
+      // If all deviceFeatures selected are in the same unit
+      // We do a average of all values
+      if (data.length > 0 && allUnitsAreSame) {
```
Comment (atrovato) [inline front/src/components/boxs/chart/Chart.jsx]: Don't you prefer to check `data.length > 0` first, then check units?
A bit more efficient (just a bit).

**Review by atrovato — COMMENTED** [review]
Looks great (on code review), just a simple comment on dupe check.

File: front/src/components/boxs/chart/Chart.jsx (hunk 1 above)
Comment (Pierre-Gilles) [reply front/src/components/boxs/chart/Chart.jsx]: I don't see what you mean? Javascript do lazy-evaluation, so if data.length is false, it won't go to the next check

Comment (atrovato) [reply front/src/components/boxs/chart/Chart.jsx]: But you still loop on "units" to check for single value, but for nothing.

Comment (Pierre-Gilles) [reply front/src/components/boxs/chart/Chart.jsx]: Ok I see! It's not a huge change, but at least i'll be more readable.

Fixed!

**Review by atrovato — APPROVED** [review]
Nice enough to me :)

**Pull request merged**